### PR TITLE
Always update image details

### PR DESCRIPTION
**Pull request by fmuyassarov**
Always update the image data within `ValidateManagementAccess()` regardless of Ironic node existence or not. Currently, image data update is happening only if Ironic node doesn't exist. That means, if we failed to write image data, we won't be able to update it again in the next retries since node already exists.
Fixes https://github.com/metal3-io/baremetal-operator/issues/839.

### CODE REVIEW

**Comment by fmuyassarov** [comment]
This should probably wait until https://github.com/metal3-io/baremetal-operator/pull/837 lands ?

**Comment by zaneb** [comment]
> This should probably wait until #837 lands ?

One of us will have to rebase, for sure :)

This looks correct to me. Some information in the commit message wouldn't go astray.
/approve

**Comment by fmuyassarov** [comment]
> > This should probably wait until #837 lands ?
> 
> One of us will have to rebase, for sure :)
> 
> This looks correct to me. Some information in the commit message wouldn't go astray.
> /approve

I'm find to put a hold on this and wait until #837 gets merged, since this is not an urgent, and rebasing 837 just because of this might bother too much.
/hold

**Comment by hardys** [comment]
#837 merged so this can be rebased and the hold removed when you get a moment @fmuyassarov :)

**Comment by fmuyassarov** [comment]
/test-integration
/hold cancel
@hardys Rebased this PR. Thanks for heads up!

File: pkg/provisioner/ironic/ironic.go
Comment (namnx228) [inline pkg/provisioner/ironic/ironic.go:537]: nit: this blank line should be deleted

**Review by namnx228 — COMMENTED** [review]
LGTM. Just a small nit

File: pkg/provisioner/ironic/ironic.go
Comment (fmuyassarov) [reply pkg/provisioner/ironic/ironic.go]: fixed, thanks.

**Comment by smoshiur1237** [comment]
Looks good to me.

**Review by dtantsur — APPROVED** [review]
/lgtm

**Pull request merged**
